superset/initialization/__init__.py

```diff
@@ -607,6 +609,8 @@ def configure_fab(self) -> None:
         appbuilder.base_template = "superset/base.html"
         appbuilder.security_manager_class = custom_sm
         appbuilder.init_app(self.superset_app, db.session)
+        custom_sm.set_user_model(SupersetUser)
```

Comment:
line 612, 613 should be declared before 610, you should assign the security_manager after it is ready to be used. 
and I would extract it to different method